src/axios-har-tracker.ts

```diff
@@ -1,112 +1,204 @@
+          wait: 10,
+          receive: 10,
+          _blocked_queueing: -1
+        }
+      }
+      return responseObject;
     }
-  );
-
-  axios.interceptors.response.use(
-    async resp => {
-      if (resp) {
-        newEntry.response = {
-          status: resp.status,
-          statusText: resp.statusText,
-          headers: getHeaders(resp.headers),
-          startedDateTime: new Date(resp.headers.date),
-          time: resp.headers['request-duration'] = Math.round(
-            process.hrtime(resp.headers['request-startTime'])[0] * 1000 +
-              process.hrtime(resp.headers['request-startTime'])[1] / 1000000
-          ),
-          httpVersion: `HTTP/${resp.request.res.httpVersion}`,
-          cookies: getCookies(JSON.stringify(resp.config.headers['Cookie'])),
-          bodySize: JSON.stringify(resp.data).length,
-          redirectURL: '',
-          headersSize: -1,
-          content: {
-            size: JSON.stringify(resp.data).length,
-            mimeType: resp.headers['content-type'] ? resp.headers['content-type'] : 'text/plain',
-            text: JSON.stringify(resp.data)
-          },
-          cache: {},
-          timings: {
-            blocked: -1,
-            dns: -1,
-            ssl: -1,
-            connect: -1,
-            send: 10,
-            wait: 10,
-            receive: 10,
-            _blocked_queueing: -1
-          }
-        };
-        const enteriesContent = Object.assign({}, newEntry);
-        generatedHar.log.entries.push(enteriesContent);
+  }
+
+  private pushNewEntryResponse(response) {
+    this.newEntry.response = this.responseObject(response);
+    this.generatedHar.log.entries.push(this.newEntry);
+  }
+
+  private pushNewEntryRequest(request) {
+    this.newEntry.request = this.requestObject(request);
+    this.generatedHar.log.entries.push(this.newEntry);
+    console.log("DEBUG this.newEntry.request:", this.newEntry.request)
+  }
+
+  constructor(requestModule: AxiosStatic) {
+    this.axios = requestModule;
+
+    this.generatedHar = {
+      log: {
+        version: '1.2',
+        creator: {
+          name: 'axios-har-tracker',
+          version: '1.0.0'
+        },
+        pages: [],
+        entries: []
+      }
+    };
+
+    this.axios.interceptors.request.use(
+
+      async config => {
+        this.generateNewEntry();
+        this.newEntry.request = this.requestObject(config);
+        return config;
+      },
+      async error => {
+        this.pushNewEntryRequest(error.request);
+        return Promise.reject(error);
+      }
+    );
+
+    this.axios.interceptors.response.use(
+      async resp => {
+        this.newEntry.response = this.responseObject(resp);
+        this.generatedHar.log.entries.push(this.newEntry);
         return resp;
+      },
+      async error => {
+        this.pushNewEntryResponse(error.response);
+        return Promise.reject(error);
+      }
+    );
+  }
+
+  private generateNewEntry() {
+    this.newEntry = {
```

Comment:
Create newEntry as local object using const and return it from this function. Then, when resetting newEntry object just do
```js
this.newEntry = generateNewEntry();
```

Reply:
thanks